### PR TITLE
feat: US5-1-2 - poi frontend

### PR DESCRIPTION
**Pull request by FrancoisMartinez**
# POI Search Functionality Implementation Summary

This update introduces a complete, end-to-end implementation for Point of Interest (POI) searching across campuses, including backend services, frontend components, and state management logic.

---

## Backend Changes
* **Refactoring:** Moved `GooglePlaceMatchUtil` from the `service` package to the `util` package for better architectural alignment.
* **Service Enhancement:** Updated `PointOfInterestService` with robust filtering logic, including null checks and string trimming for cleaner data processing.
* **Testing:** Added `PointOfInterestServiceTests`, a comprehensive test suite (225+ lines) covering various edge cases.
* **Data:** Introduced `poi_cache.json` containing 21 sample locations across SGW and Loyola campuses.

---

## Frontend - New Components
* **`BaseMarkerIcon`**: A reusable SVG base for map markers, shared between buildings and POIs.
* **`PoiMarker`**: Specific markers that render category-appropriate icons (Café, Restaurant, Pharmacy, Bar, Grocery).
* **`PoiResultsPanel`**: A slide-up UI component that displays search results and provides quick actions for directions.
* **`PoiService`**: API integration layer for fetching POI data from the backend.

---

## Frontend - New Hooks & Utilities
* **`useMarkerCallout`**: Centralized logic for handling native map callouts (refactored out of `BuildingMarker`).
* **`distanceUtils`**: Implements the Haversine formula for calculating distances to support radius-based filtering.
* **`poiUtils`**: Utility to convert POI objects into Building objects to maintain a consistent directions workflow.

---

### Reducers & Hooks
* **`mapUIReducer`**: Added 6 new actions for the POI lifecycle:
    * `POI_LOADING`, `POI_LOADED`, `POI_ERROR`, `SELECT_POI`, `CLEAR_POI`, `BACK_TO_SEARCH`.
* **`searchPanelReducer`**: Added state for radius filtering (`radiusKm`, `radiusDropdownOpen`).
* **`useMapUI`**: Integrated fetch/filter logic and added callbacks for selecting and clearing POIs.

### UI Integration
* **`SearchPanel`**: Added a radius selection dropdown (1km, 2km, 5km, or no limit).
* **`GoogleMaps`**: Updated to handle the rendering of POI markers and the results panel overlay.

---

## UI/UX Enhancements
* **Category Selector**: Users can now filter searches by category (Café, Restaurant, Pharmacy, etc.).
* **Rich Results**: The results panel displays the POI address, a location pin, and a directions icon.
* **Interaction**: Automatic callout display when a POI is selected from the list.
* **Contextual Info**: The `DirectionPanel` now dynamically displays the POI description or the building code depending on the destination type.

### CODE REVIEW

File: mobile/src/hooks/useMapUI.ts [hunk 1]
```diff
@@ -36,12 +39,16 @@ export function useMapUI(
   );
 
   // ── Derived user coordinates ──
-  const userCoords: UserCoords | null = location
-    ? {
-        latitude: location.coords.latitude,
-        longitude: location.coords.longitude,
-      }
-    : null;
+  const userCoords: UserCoords | null = useMemo(
```
Comment (Shamsam) [inline mobile/src/hooks/useMapUI.ts]: Interesting. Why did we have to use memo here?
